docker-compose.yml

```diff
@@ -0,0 +1,169 @@
+
+    # NOTE: `depends_on` only waits for the services listed below to start; it doesn't wait for them
+    # to actually be "ready" (whatever "ready" means for a particular service). If you need to
+    # control the startup of services more granularly, use the `wait-for-it.sh` script available in
+    # this image. The reason why we're not waiting for `db` service to actually be ready here is
+    # because we're waiting for the `migration` service to start, and `migration` service won't
+    # start until the `db` service is ready.
+    depends_on:
+      - db
+      - migration
+      - nginx_proxy
+      - proxy_gen
+      - redis
+
+  nginx_proxy:
+    image: nginx:stable-alpine
+    container_name: nginx_proxy
+    restart: always
+    ports:
+      - '80:80'
+      - '443:443'
+    labels:
+      - 'com.github.jrcs.letsencrypt_nginx_proxy_companion.nginx_proxy'
+    volumes:
+      - certs:/etc/nginx/certs:ro
+      - nginx_conf:/etc/nginx/conf.d
+      - nginx_dhparam:/etc/nginx/dhparam
+      - nginx_html:/usr/share/nginx/html
+      - nginx_vhost:/etc/nginx/vhost.d
+
+  # This generates reverse proxy configurations for nginx, setup taken from:
+  # https://github.com/nginx-proxy/docker-letsencrypt-nginx-proxy-companion/blob/master/docs/Docker-Compose.md
+  # (the three container setup)
+  proxy_gen:
+    image: jwilder/docker-gen:0.7.0
+    restart: always
+    command: -notify-sighup nginx_proxy -watch /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf
+    environment:
+      - ENABLE_IPV6=true
+    labels:
+      - 'com.github.jrcs.letsencrypt_nginx_proxy_companion.docker_gen'
+    volumes:
+      - certs:/etc/nginx/certs:ro
+      - nginx_conf:/etc/nginx/conf.d
+      - nginx_dhparam:/etc/nginx/dhparam
+      - nginx_html:/usr/share/nginx/html
+      - nginx_vhost:/etc/nginx/vhost.d
+      - ./server/deployment/docker-nginx.tmpl:/etc/docker-gen/templates/nginx.tmpl:ro
+      - /var/run/docker.sock:/tmp/docker.sock:ro
+    depends_on:
+      - nginx_proxy
+
+  letsencrypt:
+    image: jrcs/letsencrypt-nginx-proxy-companion:stable
+    restart: always
+    environment:
+      - DEFAULT_EMAIL=${SB_LETSENCRYPT_EMAIL}
+    volumes:
+      - certs:/etc/nginx/certs:rw
+      - nginx_html:/usr/share/nginx/html
+      - nginx_vhost:/etc/nginx/vhost.d
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+    depends_on:
+      - nginx_proxy
+      - proxy_gen
+
+  redis:
+    image: redis:5.0.5-alpine
+    restart: always
+    command: ['redis-server', '--appendonly', 'yes']
+    volumes:
+      - redis_data:/data
+
+  db:
+    image: postgres:12-alpine
+    restart: always
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=${POSTGRES_SUPER_PASSWORD}
+      - POSTGRES_DB=postgres
+      - SB_DB=${SB_DB:-shieldbattery}
+      - SB_DB_USER=${SB_DB_USER:-shieldbattery}
+      - SB_DB_PASSWORD=${SB_DB_PASSWORD}
+    volumes:
+      - ./server/deployment/db/init-scripts:/docker-entrypoint-initdb.d:ro
```

Comment:
This would ideally be provided from `app_server`'s filesystem, might be a weird order of operations here though, given the dependencies? Need to think about it more